Makefile

```diff
@@ -12,28 +13,37 @@ build: core $(all_apps)
 .PHONY: clean
 clean: clean-core $(addprefix clean-app-,$(apps))
 
+node_modules: package.json package-lock.json
+	npm install && touch node_modules
+
+core/css/uikit.%.css: src/themes/%.less node_modules
+	node_modules/less/bin/lessc src/themes/$*.less core/css/uikit.$*.css --relative-urls
+
+core/js/core.bundle.js: node_modules
+	npm run build
+
 #
 # core
 #
 .PHONY: core
-core:
-	npm install
-	node_modules/less/bin/lessc src/themes/owncloud.less core/css/uikit.owncloud.css --relative-urls
-	npm run build
+core: core/js/core.bundle.js core/css/uikit.owncloud.css
```

Comment:
building core requires the core bundle and the uikit css file.

"uikit.owncloud.css" matches the rule "uikit.%.css" which replaces % (and $*) with "owncloud".
if we have more such files in the future and require them in the core build, they'd also match automatically